tests/macros/utils.js

```diff
@@ -183,20 +341,55 @@ function lintHTML(html, fragment = true) {
         });
         return null;
     } catch (error) {
-        const error_message = error.message
+        const error_message = /** @type {Error} */ (error).message
             .split(os.EOL)
             .filter(line => line.startsWith('Error: '))
             .join(os.EOL);
         return error_message;
     }
 }
 
+/**
+ * Reads a generic fixture file from the `fixtures` directory.
+ *
+ * @param {string|string[]} filePath
+ *        A path to a file relative to the `fixtures` directory.
+ * @param {string|{encoding?:string,flag?:string}} [options]
+ *        Either the encoding for the result, or an object that contains the encoding
+ *        and an optional flag. If a flag is not provided, it defaults to `'r'`.
+ *
+ * @returns {string|Buffer}
+ */
+function readFixture(filePath, options) {
+    if (!Array.isArray(filePath)) {
+        filePath = [filePath];
+    }
+    let absolutePath = path.resolve(__dirname, 'fixtures', ...filePath);
+    return fs.readFileSync(absolutePath, options);
+}
+
+/**
+ * Reads a JSON fixture file from the `fixtures` directory.
+ *
+ * @param {...string} filePath
+ *        A path to a file relative to the `fixtures` directory.
+ */
+function readJSONFixture(...filePath) {
+    let fileName = filePath.pop();
+    if (!path.extname(fileName)) {
+        fileName += '.json';
+    }
+    return JSON.parse(String(readFixture([...filePath, fileName])));
```

Comment:
nit: it would be more efficient to push the filename back onto the array than it is to allocate and fill a new array with ...